### PR TITLE
Reproducible images by supplying latents to pipeline

### PR DESCRIPTION
**Pull request by pcuenca**
I created a simple notebook to demonstrate how it works.

TODO:
- [x] Adapt the notebook to Colab (pip install, number of samples).

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py [hunk 1]
```diff
@@ -98,12 +99,16 @@ def __call__(
...
+        # get the initial random noise
+
+        latents_shape = (batch_size, self.unet.in_channels, height // 8, width // 8)
+        if latents is None:
+            latents = torch.randn(
+                latents_shape,
+                generator=generator,
+                device=self.device,
+            )
+        assert latents.shape == latents_shape, f"Unexpected latents shape, got {latents.shape}, expected {latents_shape}"
```
Comment (patil-suraj) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py:111]: Let's raise `ValueError` asserts are meant for debugging.

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py [hunk 2]
```diff
@@ -98,12 +99,16 @@ def __call__(
...
-        # get the intial random noise
-        latents = torch.randn(
-            (batch_size, self.unet.in_channels, height // 8, width // 8),
-            generator=generator,
-            device=self.device,
-        )
+        # get the initial random noise
+
+        latents_shape = (batch_size, self.unet.in_channels, height // 8, width // 8)
+        if latents is None:
```
Comment (patil-suraj) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py:105]: if `latents` are provided  we should  put them on `self.device`.

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py [hunk 3]
```diff
@@ -46,6 +46,7 @@ def __call__(
         guidance_scale: Optional[float] = 7.5,
         eta: Optional[float] = 0.0,
         generator: Optional[torch.Generator] = None,
+        latents: Optional[torch.Tensor] = None,
```
Comment (patil-suraj) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py:49]:
```suggestion
        latents: Optional[torch.FloatTensor] = None,
```

**Review by patil-suraj — APPROVED** [review]
LGTM! Maybe also add a test for this.

**Comment by patil-suraj** [comment]
Let's not put notebook in the repo , as this would make it bukly.  Example notebooks we could go in `huggingface/notebooks` repo, or on personal github and we could link to it from the readme.

**Comment by pcuenca** [comment]
> Let's not put notebook in the repo

Moved to personal repo :)

**Review by patrickvonplaten — APPROVED** [review]
LGTM!

**Review by anton-l — APPROVED** [review]
Great addition!
+1 for a quick test as suggested by @patil-suraj, but maybe after we upload a dummy SD model (these take a while to run already).

**Pull request merged**